test/test_suite_test.go

```diff
@@ -55,11 +64,57 @@ func TestTest(t *testing.T) {
 		t, ginkgoext.GetScopeWithVersion(), []Reporter{junitReporter})
 }
 
+func goReportVagrantStatus() chan bool {
+	if ginkgoconfig.DefaultReporterConfig.Verbose ||
+		ginkgoconfig.DefaultReporterConfig.Succinct {
+		// Dev told us they want more/less information than default. Skip.
+		return nil
+	}
+
+	exit := make(chan bool)
+	go func() {
+		done := false
+		iter := 0
+		for {
+			var out string
+
+			select {
+			case ok := <-exit:
+				if ok {
+					out = "●\n"
+				} else {
+					out = "◌\n"
+				}
+				done = true
+			default:
+				out = string(rune(int('◜') + iter%4))
+			}
+			fmt.Printf("\rSpinning up vagrant VMs... %s", out)
+			if done {
+				return
+			}
+			time.Sleep(250 * time.Millisecond)
+			iter++
+		}
+	}()
+	return exit
+}
+
 var _ = BeforeSuite(func() {
-	scope := ginkgoext.GetScope()
-	switch scope {
+	var err error
+
+	if !reprovision {
+		// The developer has explicitly told us that they don't care
+		// about updating Cilium inside the guest, so skip setup below.
+		return
+	}
+
+	progressChan := goReportVagrantStatus()
+	defer func() { progressChan <- err == nil }()
```

Comment:
And close the `progressChan` channel here as well

Reply:
I don't think we can close it from both ends, either.